executor/src/witgen/identity_processor.rs

```diff
@@ -205,8 +205,14 @@ impl<'a, 'b, 'c, T: FieldElement, Q: QueryCallback<T>> IdentityProcessor<'a, 'b,
         let selector_value = right
             .selector
             .as_ref()
-            .map(|s| current_rows.evaluate(s).unwrap().constant_value().unwrap())
-            .unwrap_or(T::one());
+            .map(|s| {
+                current_rows
+                    .evaluate(s)
+                    .ok()
+                    .and_then(|affine_expression| affine_expression.constant_value())
+                    .ok_or(EvalError::Generic("Selector is not 1!".to_string()))
+            })
+            .unwrap_or(Ok(T::one()))?;
```

Comment:
Turns a panic into an error. This happens in the `block_lookup_or_permutation` test:
- A block is computed by a permutation using `selector2`
- A loop is detected
- When checking the next identity using `selector1`, `process_link()` on the last row. But `selector1` is unknown at this point.
- The error is caught by the loop detection code and the executor reverts to the normal solving algorithm.